pydantic/main.py

```diff
@@ -981,6 +981,8 @@ def validate_model(  # noqa: C901 (ignore complexity)
             if field.required:
                 errors.append(ErrorWrapper(MissingError(), loc=field.alias))
                 continue
+            elif config.partial:
+                continue
```

Comment:
This is kind of the cruz of the question about partial and how it should behave.

The problem is that with this (or any implementation of partial/non-total) the runtime instance no-longer matches the definition of the class in code.

I can imagine people legitimating wanting this to work both ways.

What if we added another config property `config.partial_omit`, then based on that, either omitted the attribute or made it `None`?

I think it would also be smart if we could change the `AttributeError` when accessing omitted properties to something like

    AttributeError: 'MyModel' object has no attribute 'bar'  - this is a partial model and the field was omitted

But only if we can do this easily and performantly.